File: nagasena/src/org/openexi/proc/io/ValueScanner.java
package org.openexi.proc.io;

import java.io.IOException;
import java.io.InputStream;

import org.openexi.proc.grammars.ValueApparatus;
import org.openexi.schema.Characters;

public abstract class ValueScanner extends ValueApparatus {
  
  protected InputStream m_istream;

  protected ValueScanner() {
    m_istream = null;
  }
  
  public abstract int getBuiltinRCS(int simpleType);
  
  public abstract void setInputStream(InputStream istream);
  
  public abstract Characters scan(int name, int uri, int tp) throws IOException;

}

File: nagasenacs/src/Nagasena/Proc/IO/ValueScanner.cs
using System.IO;
using ValueApparatus = Nagasena.Proc.Grammars.ValueApparatus;
using Characters = Nagasena.Schema.Characters;

namespace Nagasena.Proc.IO {

  /// <exclude/>
  public abstract class ValueScanner : ValueApparatus {

    protected internal Stream m_istream;

    protected internal ValueScanner() {
      m_istream = null;
    }

    public abstract int getBuiltinRCS(int simpleType);

    public abstract Stream InputStream { set; }

    public abstract Characters scan(int name, int uri, int tp);

  }

}